### PR TITLE
add maven resolver via jboss shrinkwrap

### PR DESCRIPTION
**Pull request by deusaquilus**
Fixing #523 by implementing @kgignatyev's solution of $maven resolver via shrinkwrap which correctly fetches maven dependencies.

### CODE REVIEW

**Comment by deusaquilus** [comment]
Including #547 and #524, specifically this is an implementation of #524.

**Comment by deusaquilus** [comment]
Hey @lihaoyi,
Any chance you can have a look at this?

File: build.sbt [hunk 1]
```diff
@@ -27,11 +27,24 @@ val macroSettings = Seq(
 )
 
 val sharedSettings = Seq(
-
+  assemblyMergeStrategy in assembly := {
+    case PathList("javax", "servlet", xs@_*) => MergeStrategy.first
+    case m if m.toLowerCase.endsWith("manifest.mf") => MergeStrategy.discard
+    case m if m.toLowerCase.matches("meta-inf.*\\.sf$") => MergeStrategy.discard
+    case _ => MergeStrategy.first
```
Comment (jhnsmth) [inline build.sbt]: I'd like to make a guess that the above line is guilty for failures in CI. Also, the previous two line are just the same as in the default merge strategy.  According to the sbt-assembly documentation you should use the default strategy for all cases that don't need special treatment.  Could you please tell what was the original merge conflict you tried to resolve ?

**Comment by lihaoyi** [comment]
This is obsoleted by 662859f84f5a77efca3e7e4d3074626570a992cf

Also, the test you provided passes reliably on master, at least on my machine. Given that, I have no idea what all this code is doing?

```scala
lihaoyi Ammonite$ amm
Loading...
Welcome to the Ammonite Repl 0.8.5
(Scala 2.12.2 Java 1.8.0_112)
If you like Ammonite, please support our development at www.patreon.com/lihaoyi
@ import $$maven.`com.github.nscala-time::nscala-time:2.16.0`
Import Hook could not be resolved
@ import $maven.`com.github.nscala-time::nscala-time:2.16.0`
Import Hook could not be resolved
@ import $ivy.`com.github.nscala-time::nscala-time:2.16.0`
:: loading settings :: url = jar:file:/usr/local/bin/amm!/org/apache/ivy/core/settings/ivysettings.xml
:: resolving dependencies :: com.github.nscala-time#nscala-time_2.12-caller;working
	confs: [default]
	found com.github.nscala-time#nscala-time_2.12;2.16.0 in central
	found org.scala-lang#scala-library;2.12.1 in chain-resolver
	[2.12.1] org.scala-lang#scala-library;2.12.1
	found joda-time#joda-time;2.9.7 in central
	found org.joda#joda-convert;1.2 in chain-resolver
downloading http://repo1.maven.org/maven2/com/github/nscala-time/nscala-time_2.12/2.16.0/nscala-time_2.12-2.16.0.jar ...
	[SUCCESSFUL ] com.github.nscala-time#nscala-time_2.12;2.16.0!nscala-time_2.12.jar (677ms)
downloading http://repo1.maven.org/maven2/com/github/nscala-time/nscala-time_2.12/2.16.0/nscala-time_2.12-2.16.0-sources.jar ...
	[SUCCESSFUL ] com.github.nscala-time#nscala-time_2.12;2.16.0!nscala-time_2.12.jar(source) (221ms)
downloading http://repo1.maven.org/maven2/com/github/nscala-time/nscala-time_2.12/2.16.0/nscala-time_2.12-2.16.0-javadoc.jar ...
	[SUCCESSFUL ] com.github.nscala-time#nscala-time_2.12;2.16.0!nscala-time_2.12.jar(javadoc) (1020ms)
downloading http://repo1.maven.org/maven2/joda-time/joda-time/2.9.7/joda-time-2.9.7.jar ...
	[SUCCESSFUL ] joda-time#joda-time;2.9.7!joda-time.jar (292ms)
import $ivy.$
@ import com.github.nscala_time.time.Imports._
import com.github.nscala_time.time.Imports._
@
```

**Pull request closed without merging**